src/main/java/com/mongodb/jdbc/SortableBsonDocument.java

```diff
@@ -0,0 +1,61 @@
+package com.mongodb.jdbc;
+
+import java.util.List;
+import org.bson.BsonDocument;
+
+public class SortableBsonDocument extends BsonDocument implements Comparable<SortableBsonDocument> {
+
+    static class SortSpec {
+        String key;
+        ValueType type;
+
+        SortSpec(String key, ValueType type) {
+            this.key = key;
+            this.type = type;
+        }
+    }
+
+    enum ValueType {
+        String,
+        Int,
+    }
+
+    List<SortSpec> sortSpecs;
+    BsonDocument nestedDocValue;
+
+    SortableBsonDocument(List<SortSpec> sortSpecs, String key, BsonDocument docValue) {
+        super(key, docValue);
+
+        this.sortSpecs = sortSpecs;
+        this.nestedDocValue = docValue;
+    }
+
+    @Override
+    public int compareTo(SortableBsonDocument o) {
+        int r = 0;
+        for (SortSpec sortSpec : this.sortSpecs) {
+            switch (sortSpec.type) {
+                case String:
```

Comment:
Note that we need to use `getString` or `getInt32` on the `BsonDocument` to get the relevant sort field. That is the only reason we have this type enum and switch on it. The `BsonValue` class returned by the generic `get` method does not implement `Comparable`, unfortunately. Open to other suggestions on how to handle sorting but I think this is a reasonable alternative to the massive in-line `Comparator`s I had last time around.